src/webapp/components/form/useForm.ts

```diff
@@ -0,0 +1,34 @@
+import { useCallback, useEffect, useState } from "react";
+
+import { Form, FormState } from "./FormState";
+import { FormFieldState } from "./FormFieldState";
+
+export function useForm(
+    formState: FormState,
+    onFormChange: (newFormState: FormState, updatedField: FormFieldState) => void
+): {
+    formLocalState: FormState;
+    handleUpdateFormField: (updatedField: FormFieldState) => void;
+} {
+    const [formLocalState, setFormLocalState] = useState<FormState>(formState);
```

Comment:
Isn't [formLocalState, setFormLocalState] an exact copy of [formState, setFormState]  in useDiseaseOutbreakEventForm. Why do we need 2 states to hold the same info? 
Also, in react when we want to update a state based on previous sate, we need to use callback inside setState otherwise it causes issues. I have a weird feeling we are going to run into this issue, but am not able to pinpoint the place where this could potentially happen. Refer : https://react.dev/reference/react/useState#updating-state-based-on-the-previous-state.

Reply:
ah, very interesting idea! I will understand it better whenever we meet next :)